research/2d-decoding/rectangle_detector.py

```diff
@@ -103,42 +114,47 @@ def readImage(filename):
         nottaSquare = False
         # Find the pixels with the biggest and smallest x and y values, if more than 1 pixel have the smallest x, then take the one with the smallest y
         # Biggest y we take the smallest x, biggest x we take the biggest y and smallest y we take the biggest x (basically rotating clockwise)
-        minX, maxX = min(mins(group, key = lambda a: a[0]), key = lambda a: a[1]), max(maxs(group, key = lambda a: a[0]), key=lambda a: a[1])
-        minY, maxY = max(mins(group, key = lambda a: a[1]), key=lambda a: a[0]), min(maxs(group, key = lambda a: a[1]), key = lambda a: a[0])
+        minX, maxX = min(mins(group, key=lambda a: a[0]), key=lambda a: a[1]), max(
+            maxs(group, key=lambda a: a[0]), key=lambda a: a[1])
+        minY, maxY = max(mins(group, key=lambda a: a[1]), key=lambda a: a[0]), min(
+            maxs(group, key=lambda a: a[1]), key=lambda a: a[0])
 
         # Check that all the points are different
         if not minX != maxX != minY != maxY:
             continue
 
         # Line equations are y = m(x-a)+b, with A(a, b) being on the line and m being the slope
         try:
-            line = lambda a: (maxY[1]-maxX[1])/(maxY[0]-maxX[0])*(a - maxX[0]) + maxX[1] # Lower right side
+            # Lower right side
+            def line(a): return (maxY[1]-maxX[1]) / \
+                (maxY[0]-maxX[0])*(a - maxX[0]) + maxX[1]
```

Comment:
Why not lambda?
Also, confusing lol, comment plzzzz
Also also, when breaking equation lines, the operator should go at the start of the new line (to avoid stuff like `/ \` lol)
ie.
```py
def line(a): return (maxY[1]-maxX[1])  \
                / (maxY[0]-maxX[0])*(a - maxX[0]) + maxX[1]
```

Reply:
Outdated
Equation line question: see [Pep8 ](https://peps.python.org/pep-0008/#should-a-line-break-before-or-after-a-binary-operator )